src/layers/core/path-layer/path-layer-vertex.glsl

```diff
@@ -1,71 +1,197 @@
 #define SHADER_NAME path-layer-vertex-shader
 
-attribute float directions;
 attribute vec3 positions;
-attribute vec3 leftDeltas;
-attribute vec3 rightDeltas;
 
-attribute vec4 colors;
-attribute vec3 pickingColors;
+attribute vec3 instanceStartPositions;
```

Comment:
We have always been calculating the line joints in the vertex shader. The layer itself is projection agnostic. I share your concern about the if statements. As the title suggests this is a quick POC, and there's a lot of room for optimization.